src/Common/NamedCollections/NamedCollections.h

```diff
@@ -51,6 +51,11 @@ class NamedCollection
 
     template <typename T, bool locked = false> void setOrUpdate(const Key & key, const T & value);
 
+    bool isOverridable(const Key & key, bool default_value) const;
+
+    template <bool locked = false>
+    void setOverridable(const Key & key, bool value);
```

Comment:
I'd like to remove this method and add an aditional `is_overridable` argument in `set` and `setOrUpdate` methods instead because it will make sure we do not forget to add this flag in all places where we set values in named collection.